MUI/src/main/java/mui/MUIStateListProvider.java

```diff
@@ -0,0 +1,142 @@
+
+	private static int maxStateId;
+	private static HashSet<Integer> numsSent;
+
+	public MUIStateListProvider(PluginTool tool, String name) {
+		super(tool, name, name);
+		buildStateListView();
+		buildMainPanel();
+		setTitle("MUI State List");
+		setDefaultWindowPosition(WindowPosition.RIGHT);
+		setVisible(false);
+	}
+
+	private void buildMainPanel() {
+		mainPanel = new JPanel(new BorderLayout());
+		mainPanel.add(stateListView, BorderLayout.CENTER);
+	}
+
+	private void buildStateListView() {
+		rootNode =
+			new DefaultMutableTreeNode("States");
+
+		activeNode = new DefaultMutableTreeNode("Active");
+		waitingNode = new DefaultMutableTreeNode("Waiting");
+		forkedNode = new DefaultMutableTreeNode("Forked");
+		completeNode = new DefaultMutableTreeNode("Complete");
+		erroredNode = new DefaultMutableTreeNode("Errored");
+
+		rootNode.add(activeNode);
+		rootNode.add(waitingNode);
+		rootNode.add(forkedNode);
+		rootNode.add(completeNode);
+		rootNode.add(erroredNode);
+
+		treeModel = new DefaultTreeModel(rootNode);
+
+		stateListTree = new JTree(treeModel);
+		stateListView = new JScrollPane(stateListTree);
+	}
+
+	public static void tryUpdate(ManticoreRunner runner, Boolean force) {
+
+		if (force || runner == runnerDisplayed) {
+
+			maxStateId = 0;
+			numsSent = new HashSet<Integer>();
+			numsSent.clear();
+
+			ManticoreStateListModel stateListModel = runner.stateListModel;
+			activeNode.removeAllChildren();
+			waitingNode.removeAllChildren();
+			forkedNode.removeAllChildren();
+			completeNode.removeAllChildren();
+			erroredNode.removeAllChildren();
+			try {
+				stateListModel.stateList.get(StateOuterClass.State.StateType.BUSY)
+						.forEach((st) -> activeNode.add(stateToNode(st)));
+				stateListModel.stateList.get(StateOuterClass.State.StateType.READY)
+						.forEach((st) -> waitingNode.add(stateToNode(st)));
+				stateListModel.stateList.get(StateOuterClass.State.StateType.KILLED)
+						.forEach((st) -> erroredNode.add(stateToNode(st)));
+				stateListModel.stateList.get(StateOuterClass.State.StateType.TERMINATED)
+						.forEach((st) -> completeNode.add(stateToNode(st)));
+
+				activeNode.setUserObject(String.format("Active (%d)", activeNode.getChildCount()));
+				waitingNode
+						.setUserObject(String.format("Waiting (%d)", waitingNode.getChildCount()));
+				completeNode.setUserObject(
+					String.format("Complete (%d)", completeNode.getChildCount()));
+				erroredNode
+						.setUserObject(String.format("Errored (%d)", erroredNode.getChildCount()));
+
+				for (int i = 1; i <= maxStateId; i++) {
+					if (!numsSent.contains(i)) {
+						forkedNode.add(new DefaultMutableTreeNode(String.format("State %d", i)));
+					}
+				}
+				forkedNode.setUserObject(String.format("Forked (%d)", forkedNode.getChildCount()));
+
+				rootNode.setUserObject(String.format("States (%d)", maxStateId));
+			}
+			catch (NullPointerException npe) {
+				Msg.info(stateListModel, "no states yet");
+			}
```

Comment:
I think we should avoid catching NullPointerExceptions. Instead we should explicitly check if the thing we think is going to be null _is_ null and then write our message. What are you expecting to be null when there are no states yet?